cmd/zc_traverser_blob_account.go

```diff
@@ -51,68 +51,81 @@ func (t *blobAccountTraverser) IsDirectory(_ bool) bool {
 	return true // Returns true as account traversal is inherently folder-oriented and recursive.
 }
 
-func (t *blobAccountTraverser) listContainers() ([]string, error) {
+func (t *blobAccountTraverser) listContainers() (chan string, chan error) {
+	containers := make(chan string, 5000)
+	e := make(chan error)
+
 	// a nil list also returns 0
-	if len(t.cachedContainers) == 0 {
+	go func() {
+		var err error = nil
+		defer func() {
+			close(containers)
+			e <- err
+			close(e)
+		}()
+
+		if len(t.cachedContainers) != 0 {
+			for _, v := range t.cachedContainers {
+				containers <-v
+			}
+			return
+		}
+
 		marker := azblob.Marker{}
 		cList := make([]string, 0)
 
 		for marker.NotDone() {
 			resp, err := t.accountURL.ListContainersSegment(t.ctx, marker, azblob.ListContainersSegmentOptions{})
 
 			if err != nil {
-				return nil, err
+				return
 			}
 
 			for _, v := range resp.ContainerItems {
 				// Match a pattern for the container name and the container name only.
 				if t.containerPattern != "" {
 					if ok, err := containerNameMatchesPattern(v.Name, t.containerPattern); err != nil {
 						// Break if the pattern is invalid
-						return nil, err
+						return
 					} else if !ok {
 						// Ignore the container if it doesn't match the pattern.
 						continue
 					}
 				}
 
+				containers <- v.Name
 				cList = append(cList, v.Name)
 			}
 
 			marker = resp.NextMarker
 		}
 
 		t.cachedContainers = cList
+		return
+	}()
 
-		return cList, nil
-	} else {
-		return t.cachedContainers, nil
-	}
+	return containers, e
 }
 
 func (t *blobAccountTraverser) Traverse(preprocessor objectMorpher, processor objectProcessor, filters []ObjectFilter) error {
 	// listContainers will return the cached container list if containers have already been listed by this traverser.
-	cList, err := t.listContainers()
-
-	if err != nil {
-		return err
-	}
+	cList, errChan := t.listContainers()
 
-	for _, v := range cList {
+	for v := range cList {
```

Comment:
This will only run one container listing instance at once. While it will get the job done of separating the process of enumeration of blobs from containers, it will not actually parallelize the listing of blobs.

We could consider starting up multiple goroutines to handle the blob listing on individual containers; it would not increase the price on the listing operation.

Reply:
Ditto to the other account level enumerators.